src/python/WMCore/Services/WMStats/WMStatsWriter.py

```diff
@@ -126,10 +126,18 @@ def updateRequestsInfo(self, docs):
                                         fields={'general_fields': JSONEncoder().encode(doc)})
 
     def updateAgentInfo(self, agentInfo):
-        return self.couchDB.updateDocument(agentInfo['_id'], self.couchapp,
-                                           'agentInfo',
-                                           {'agent_info': JSONEncoder().encode(agentInfo)},
-                                           useBody=True)
+        """
+        replace the agentInfo document with new one.
+        """
+        try:
+            exist_doc = self.couchDB.document(agentInfo["_id"])
+            agentInfo["_rev"] = exist_doc["_rev"]
```

Comment:
so if the doc contains a _rev field, it will simply get updated (replaced) in couch and revision ++ ?